sdks/python/apache_beam/examples/complete/game/leader_board.py

```diff
@@ -120,6 +120,8 @@ def __init__(self):
 
   def process(self, elem):
     try:
+      if isinstance(elem, bytes):
```

Comment:
Can we always decode here? It is better to have a clear expectation of input arguments as much as possible on Python 3: either always encoded bytes, or always strings, but not mixing the two.

Reply:
The IT test uses pubsub as input source while the unit test currently passes strings. That's why we cannot always decode

Reply:
I am not sure I follow - could you please elaborate - what exactly in either pubsub or unit test makes it so that the input type is not consistent here?

Reply:
The code for leader_board and game_stats is used both with strings https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/complete/game/leader_board_test.py#L45 (unicode strings) and with pubsub messages: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/complete/game/leader_board_it_test.py#L146.

Reply:
Ok, the fix would then be to:
1. Expect `elem` to be a string.
2. Decode the messages as they arrive from PubSub using `| 'DecodeString' >> Map(lambda b: b.decode('utf-8')` in the example pipeline. Note that there are several example pipelines that will need this change.

Note that we used to have `ReadStringsFromPubSub` method that was deprecated: https://github.com/apache/beam/blob/a5ed104aeac91781492523808fa4d3df6f2e55f7/sdks/python/apache_beam/io/gcp/pubsub.py#L194.

We may want to revisit this deprecation in the future.

Reply:
@tvalentyn I applied this fix